File: Golang/PlusOne/PlusOne_test.go
package PlusOne

import (
	"testing"
	"reflect"
)

func TestPlusOne(t *testing.T) {

	if reflect.DeepEqual(plusOne([]int{1}),
		[]int{2}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(plusOne([]int{9}),
		[]int{1, 0}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(plusOne([]int{9, 9}),
		[]int{1, 0, 0}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(plusOne([]int{2, 8, 9, 9, 9}),
		[]int{2, 9, 0, 0, 0}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(plusOne([]int{2, 8, 8, 9}),
		[]int{2, 8, 9, 0}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/PlusOneTest.py
from unittest import TestCase

from PlusOne import PlusOne


class TestPlusOne(TestCase):
    def test_plusOne(self):
        po = PlusOne()

        self.assertEqual(po.plusOne([1]), [2])

        self.assertEqual(po.plusOne([9]), [1, 0])

        self.assertEqual(po.plusOne([9, 9]), [1, 0, 0])

        self.assertEqual(po.plusOne([2, 8, 9, 9, 9]), [2, 9, 0, 0, 0])

        self.assertEqual(po.plusOne([2, 8, 8, 9]), [2, 8, 9, 0])
